src/algebra/monoid_algebra/basic.lean

```diff
@@ -1503,6 +1529,20 @@ finset.induction_on s rfl $ λ a s has ih, by rw [prod_insert has, ih,
 
 end
 
+lemma map_domain_algebra_map (k A : Type*) {H F : Type*} [comm_semiring k] [semiring A]
+  [algebra k A] [add_monoid G] [add_monoid H] [add_monoid_hom_class F G H] (f : F) (r : k) :
+  map_domain f (algebra_map k (add_monoid_algebra A G) r) =
+    (algebra_map k (add_monoid_algebra A H)) r :=
+by simp only [function.comp_app, map_domain_single, add_monoid_algebra.coe_algebra_map, map_zero]
```

Comment:
Is `by simp only [..a long list]` preferred over `by simp`? I don't think so.

Reply:
Without the explicit list, compilation time is multiplied by 10 (it is not much either way 1s/100ms).  I'll leave the squeezed version, unless there is a strong preference for `by simp`.